Src/VimCore/CoreInterfaces.fs

```diff
@@ -4995,6 +5058,15 @@ type IVimHost =
     /// example).  This override allows them to do this processing
     abstract TryCustomProcess: textView: ITextView -> command: InsertCommand -> bool
 
+    /// Whether multi-selection is supported
+    abstract IsMultiSelectionSupported: bool
+
+    /// Get all the selected spans for the specified text view
+    abstract GetSelectedSpans: textView: ITextView -> SelectedSpan seq
+
+    /// Set all the selected spans for the specified text view
+    abstract SetSelectedSpans: textView: ITextView -> SelectedSpan seq -> unit

```

Comment:
Agreed. I was holding off on that change to avoid pushing commits while you were still reviewing. Originally I was under the impression that this had to be a shared service, and that you had to access a shared service through the vim host. Vim shared changed all that in a good way.